### PR TITLE
Add @Generated annotation

### PR DESCRIPTION
**Pull request by srnagar**
Adding a new annotation to identify methods that are generated by AutoRest.

fixes https://github.com/Azure/autorest.java/issues/1049

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * Annotation given to all methods that are generated by AutoRest. This annotation is intended to be used by
+ * the code generation tool only to identify methods that are generated. The purpose of this annotation is to find
+ * and replace all methods in a class that are generated. Methods not annotated with this annotation will not be
+ * updated when code is regenerated.
+ * <p>This annotation is expected to be used in classes that are annotated with {@link ServiceClient} only.</p>
+ */
+@Retention(SOURCE)
+@Target(METHOD)
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java:20]: Should this also be able to target properties? There are cases where additional properties may be written by hand into a generated class.

File: sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+/**
+ * Annotation given to all methods that are generated by AutoRest. This annotation is intended to be used by
+ * the code generation tool only to identify methods that are generated. The purpose of this annotation is to find
+ * and replace all methods in a class that are generated. Methods not annotated with this annotation will not be
+ * updated when code is regenerated.
+ * <p>This annotation is expected to be used in classes that are annotated with {@link ServiceClient} only.</p>
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java:17]: Do we want to make this more generic and allow it to be used on any generated model?

File: sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java:20]: Constructor is another potential target we may want to look at.

Comment (srnagar) [reply sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java]: Added `FIELD` and `CONSTRUCTOR`

File: sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java (hunk 2 above)
Comment (srnagar) [reply sdk/core/azure-core/src/main/java/com/azure/core/annotation/Generated.java:19]: There's no usecase for using this annotation in any other model today but if we find a need to do that, we can update the documentation.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**